profiling/src/internal/observation/trimmed_observation.rs

```diff
@@ -183,4 +183,33 @@ mod tests {
             assert_eq!(*b, vec![1, 2]);
         }
     }
+
+    // cargo +nightly bolero test
+    // internal::observation::trimmed_observation::tests::fuzz_trimmed_observation -T 1min
+    #[test]
+    fn fuzz_trimmed_observation() {
+        bolero::check!().with_type::<Vec<i64>>().for_each(|v| {
+            let o = ObservationLength::new(v.len());
+            {
+                let t = TrimmedObservation::new(v.clone(), o);
```

Comment:
One other check we can do is that the length of the trimmed observation is the same as o